docs/SPEC.md

```diff
@@ -205,64 +186,61 @@ Examples:
+    date.add(d: 1mo, to: date.add(d: 1d, to: 2018-02-28T00:00:00Z))   // 2018-04-01T00:00:00Z
+    date.sub(d: 1d, from: date.add(d: 2mo, to: 2018-01-01T00:00:00Z)) // 2018-02-28T00:00:00Z
+    date.add(d: 3mo, to: date.sub(d: 1d, from: 2018-01-01T00:00:00Z)) // 2018-03-31T00:00:00Z
+    date.add(d: 1mo, to: date.add(d: 1mo, to: 2018-01-31T00:00:00Z))  // 2018-03-28T00:00:00Z
+    date.add(d: 2mo, to: 2018-01-31T00:00:00Z)                        // 2018-03-31T00:00:00Z
+
+    // Addition and subtraction of durations to date times applies months and nanoseconds in that order.
+    date.add(d: 2d, to: date.add(d: 1mo, to: 2018-01-28T00:00:00Z))  // 2018-03-02T00:00:00Z
+    date.add(d: 1mo2d, to: 2018-01-28T00:00:00Z)                     // 2018-03-02T00:00:00Z
+    date.add(d: 1mo, to: date.add(d: 2d, to: 2018-01-28T00:00:00Z))  // 2018-02-28T00:00:00Z, explicit add of 2d first changes the result
+    date.add(d: 2mo2d, to: 2018-02-01T00:00:00Z)                     // 2018-04-03T00:00:00Z
+    date.add(d: 1mo30d, to: 2018-01-01T00:00:00Z)                    // 2018-03-03T00:00:00Z, Months are applied first to get February 1st, then days are added resulting in March 3 in 2018.
+    date.add(d: 1mo1d, to: 2018-01-31T00:00:00Z)                     // 2018-03-01T00:00:00Z, Months are applied first to get February 28th, then days are added resulting in March 1 in 2018.
+
+    // Multiplication and addition of durations to date times
+    date.add(d: date.scale(d:1mo, n:1), to: 2018-01-01T00:00:00Z)  // 2018-02-01T00:00:00Z
+    date.add(d: date.scale(d:1mo, n:2), to: 2018-01-01T00:00:00Z)  // 2018-03-01T00:00:00Z
+    date.add(d: date.scale(d:1mo, n:3), to: 2018-01-01T00:00:00Z)  // 2018-04-01T00:00:00Z
+    date.add(d: date.scale(d:1mo, n:1), to: 2018-01-31T00:00:00Z)  // 2018-02-28T00:00:00Z
+    date.add(d: date.scale(d:1mo, n:2), to: 2018-01-31T00:00:00Z)  // 2018-03-31T00:00:00Z
+    date.add(d: date.scale(d:1mo, n:3), to: 2018-01-31T00:00:00Z)  // 2018-04-30T00:00:00Z
 
 #### Date and time literals
 
 A date and time literal represents a specific moment in time.
 It has a date part, a time part and a time offset part.
 The format follows the RFC 3339 specification.
-The time is optional, when it is omitted the time is assumed to be midnight for the default location.
-The time_offset is optional, when it is omitted the location option is used to determine the offset.
+The time is optional, when it is omitted the time is assumed to be midnight UTC.
 
     date_time_lit     = date [ "T" time ] .
     date              = year_lit "-" month "-" day .
     year              = decimal_digit decimal_digit decimal_digit decimal_digit .
     month             = decimal_digit decimal_digit .
     day               = decimal_digit decimal_digit .
-    time              = hour ":" minute ":" second [ fractional_second ] [ time_offset ] .
+    time              = hour ":" minute ":" second [ fractional_second ] time_offset .
```

Comment:
I left the implementation unchanged in that the time offset is optional so that its easier to provide a good error message to users if it is missing.

Reply:
I'm not sure I understand. The way I read this change to the grammar, the `time_offset` was optional before, but now it is required.

Reply:
Correct now it is required, however the implementation will parse without a time offset and report the error that its missing.